src/abstractdataframe/reshape.jl

```diff
@@ -65,49 +76,74 @@ d1 = DataFrame(a = repeat([1:3;], inner = [4]),
     cnames = names(df)[id_vars]
-    insert!(cnames, 1, :value)
-    insert!(cnames, 1, :variable)
+    insert!(cnames, 1, value_name)
+    insert!(cnames, 1, measure_name)
     DataFrame(Any[Compat.repeat(_names(df)[measure_vars], inner=nrow(df)),   # variable
                   vcat([df[c] for c in measure_vars]...),                    # value
                   [Compat.repeat(df[c], outer=N) for c in id_vars]...],      # id_var columns
               cnames)
 end
-function stack(df::AbstractDataFrame, measure_var::Int, id_var::Int)
-    stack(df, [measure_var], [id_var])
+function stack(df::AbstractDataFrame, measure_var::Int, id_var::Int;
+               measure_name=:variable, value_name=:value)
+    stack(df, [measure_var], [id_var];
+          measure_name=measure_name, value_name=value_name)
+end
+function stack(df::AbstractDataFrame, measure_vars::Vector{Int}, id_var::Int;
+               measure_name=:variable, value_name=:value)
+    stack(df, measure_vars, [id_var];
+          measure_name=measure_name, value_name=value_name)
 end
-function stack(df::AbstractDataFrame, measure_vars::Vector{Int}, id_var::Int)
-    stack(df, measure_vars, [id_var])
+function stack(df::AbstractDataFrame, measure_var::Int, id_vars::Vector{Int};
+               measure_name=:variable, value_name=:value)
+    stackdf(df, [measure_var], id_vars;
+            measure_name=measure_name, value_name=value_name)
 end
-function stack(df::AbstractDataFrame, measure_var::Int, id_vars::Vector{Int})
-    stackdf(df, [measure_var], id_vars)
+function stack(df::AbstractDataFrame, measure_vars, id_vars;
+               measure_name=:variable, value_name=:value)
+    stack(df, index(df)[measure_vars], index(df)[id_vars];
+          measure_name=measure_name, value_name=value_name)
 end
-stack(df::AbstractDataFrame, measure_vars, id_vars) =
-    stack(df, index(df)[measure_vars], index(df)[id_vars])
 # no vars specified, by default select only numeric columns
-numeric_vars(df::AbstractDataFrame) = [T <: AbstractFloat || (T <: Nullable && eltype(T) <: AbstractFloat)
-                                       for T in eltypes(df)]
-function stack(df::AbstractDataFrame, measure_vars = numeric_vars(df))
+numeric_vars(df::AbstractDataFrame) =
+    [T <: AbstractFloat || (T <: Nullable && eltype(T) <: AbstractFloat)
+     for T in eltypes(df)]
+
+function stack(df::AbstractDataFrame, measure_vars = numeric_vars(df);
+               measure_name=:variable, value_name=:value)
     mv_inds = index(df)[measure_vars]
-    stack(df, mv_inds, _setdiff(1:ncol(df), mv_inds))
+    stack(df, mv_inds, _setdiff(1:ncol(df), mv_inds);
+          measure_name=measure_name, value_name=value_name)
 end
 
 """
 Stacks a DataFrame; convert from a wide to long format; see
 `stack`.
 """
-melt(df::AbstractDataFrame, id_vars::@compat(Union{Int,Symbol})) = melt(df, [id_vars])
-function melt(df::AbstractDataFrame, id_vars)
+function melt(df::AbstractDataFrame, id_vars::@compat(Union{Int,Symbol});
```

Comment:
I felt the concise function definition form was less readable in many cases once the new keyword arguments were added. I'm happy to revert this if others disagree.